bevy_asset_loader/src/loading_state.rs

```diff
@@ -317,26 +317,23 @@ where
     /// ```
     #[allow(unused_mut)]
     pub fn build(mut self, app: &mut App) {
-        app.init_resource::<AssetLoaderConfiguration<S>>();
-        {
-            let mut asset_loader_configuration = app
-                .world
-                .get_resource_mut::<AssetLoaderConfiguration<S>>()
-                .unwrap();
-            let mut loading_config = asset_loader_configuration
-                .state_configurations
-                .remove(&self.loading_state)
-                .unwrap_or_default();
-            if self.next_state.is_some() {
-                loading_config.next = self.next_state;
-            }
-            if self.failure_state.is_some() {
-                loading_config.failure = self.failure_state;
-            }
-            asset_loader_configuration
-                .state_configurations
-                .insert(self.loading_state.clone(), loading_config);
+        let mut asset_loader_configuration = AssetLoaderConfiguration::default();
+        let mut loading_config = asset_loader_configuration
+            .state_configurations
+            .remove(&self.loading_state)
+            .unwrap_or_default();
+        if self.next_state.is_some() {
+            loading_config.next = self.next_state;
+        }
+        if self.failure_state.is_some() {
+            loading_config.failure = self.failure_state;
         }
+        asset_loader_configuration
+            .state_configurations
+            .insert(self.loading_state.clone(), loading_config);
+
+        app.insert_resource(asset_loader_configuration);
```

Comment:
This will overwrite any configuration from other loading states from the same states type. That's why the original version get the configuration from the app.
The same applies to the `DynamicAssets` below.

Reply:
Yes: https://github.com/NiklasEi/bevy_asset_loader/actions/runs/4628349358/jobs/8193537263?pr=124#step:7:193

Reply:
Wow, didn't expected that. Thanks for clarification.